File: Project/Boundaries/Event.py
from datetime import datetime


class Event:
    def __init__(self, event_id: int, event_name: str, event_type: str, event_data: str, hostname, ip_address,
                 time_stamp=datetime.now()):
        self.event_id = event_id
        self.event_name = event_name
        self.event_type = event_type
        self.event_data = event_data
        self.time_stamp = time_stamp
        self.hostname = hostname
        self.ip_address = ip_address

    def __str__(self):
        return f"EventId:{self.event_id}, " \
               f"EventName:{self.event_name}, " \
               f"EventType:{self.event_type}, " \
               f"EventData:{self.event_data}, " \
               f"Hostname:{self.hostname}, " \
               f"IPAddress:{self.ip_address}, " \
               f"TimeStamp:{self.time_stamp}"

File: Web/src/Components/Events/EventsClasses/Event.js
import { makeObservable, observable } from 'mobx';

export default class Event {
    constructor({
        id,
        eventName,
        eventType,
        eventData,
        timeStamp,
        hostname,
        IPAddress,
    }) {
        this.id = id;
        this.name = eventName;
        this.type = eventType;
        this.data = eventData;
        this.timeStamp = new Date(timeStamp);
        this.hostname = hostname;
        this.ip = IPAddress;
        makeObservable(this, {
            id: observable,
            name: observable,
            type: observable,
            data: observable,
            timeStamp: observable,
            hostname: observable,
            ip: observable,
        });
    }
}
